include/form_functions.js

```diff
@@ -0,0 +1,177 @@
+
+
+//Add student form   
+//here I use this form for both creating and editing a student. So, when creating a student
+//I don't pass the id and when editing I pass the id and thats how I diff it here. Id the studentId was passed 
+//Then I assign the values to the hidden fields in the form
+// --------------------------------------------------------------------------
+function openAddStudentForm(studentId){
+
+    if(studentId){
+        const student = document.getElementById('hiddenStudent').value;
+        const myStudent = JSON.parse(student);
+
+        document.getElementById('fname').value = myStudent.FirstName;
+        document.getElementById('lname').value = myStudent.LastName;
+        document.getElementById('email').value = myStudent.Email;
+        document.getElementById('phone').value = myStudent.Phone;
+        document.getElementById('rating').value = myStudent.Rating;
+        document.getElementById('lichess').value = myStudent.Lichess;     
+                     
+    } 
+      
+    document.getElementById('addStudentForm').style.display = 'block';
+}
+
+
+function closeAddStudentForm(){
+    document.getElementById('addStudentForm').style.display = 'none';
+}
+
+//Add credits form   
+// --------------------------------------------------------------------------
+function openCreditForm(buttonclicked, credits){ 
+    document.getElementById('hiddenValue').value = buttonclicked;                             //I'm setting the id parameter to the type hidden field so I can know wich button was clicked in the Request
+    
+    if(buttonclicked == 'Subtract'){                                                 //What's happening here is that if subtract was clicked I want the max to be set to remaining credits amount
+        document.getElementById('maxCredit').max = credits;                          //so the user cannot subtract more credits that the remaining ones. So if a user has 2 credits left a refund of more than
+        document.getElementById('maxCredit').placeholder = '1 - ' + credits;         //2 credits cannot be done.This ensure that only students with remaining credits can get a refund
+    }  
+    
+    if(buttonclicked == 'Add'){                                                 //Trying out I see that id I click subtract first and then add the placeholder and max values
+        document.getElementById('maxCredit').max = '100';                            //do not go back to normal and stay with the values I set above so I'm setting them here
+        document.getElementById('maxCredit').placeholder = '1 - 100';
+    }
```

Comment:
So here, the add/subtract buttons open a form but instead of loading the current credits to be edited in the form, what I'm doing is that I show a type number field where I restrict the max number and placeholder depending on the remaining credits the student has. The DB is updated when the form is submitted, if it was subtracted the change will go to the refund table, and if added will go to the credits table. 
I'm trying to see how to use AJAX request here. 
To get the credits through AJAX instead of passing it to the function as parameter? 
Or maybe I can have + and - buttons that do not open a form but that add/subtract credits using AJAX and the credits amount is updated it without reloading the page?

Reply:
Ah okay, I didn't realize it opened a form. That's totally fine if you prefer that UX, I just wanted to make sure there was something that communicated with the DB and it wasn't a front-end only update that was happening.